tutorials/llm/llama/domain-adaptive-pretraining/code/domain_adaptive_pretraining_nemo2.0.ipynb

```diff
@@ -0,0 +1,553 @@
+    "                        except json.JSONDecodeError:\n",
+    "                            continue\n",
+    "                if has_code:\n",
+    "                    code_files.append(file_path)\n",
+    "                    cnt_code = cnt_code + 1\n",
+    "                if has_text:\n",
+    "                    text_files.append(file_path)\n",
+    "                    cnt_text = cnt_text + 1\n",
+    "    return code_files, text_files, cnt_code, cnt_text\n",
+    "\n",
+    "# Modify data path to point to jsonl data source, in this case data_path='code/data/all_jsonl_data'\n",
+    "data_path = '/workspace/dapt-custom-tokenization/code/data/all_jsonl_data'\n",
+    "\n",
+    "code_files, text_files, cnt_code, cnt_text = identify_jsonl_files(data_path)\n",
+    "\n",
+    "print(\"\\nNumber of Files containing 'file_type':'text':\", cnt_text)\n",
+    "print(\"Number of Files containing 'file_type':'code':\", cnt_code)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "60987ff2",
+   "metadata": {},
+   "source": [
+    "### Merging code JSONL files into a single JSONL file for further preprocessing"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c02f2e6f",
+   "metadata": {},
+   "source": [
+    "This is an optional step, it is possible to use multiple jsonl files in this workflow as well. This example uses a single merged. jsonl file"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "892f4493",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import json\n",
+    "\n",
+    "def list_jsonl_files(directory):\n",
+    "    jsonl_files = []\n",
+    "    for root, _, files in os.walk(directory):\n",
+    "        for file in files:\n",
+    "            if file.endswith('.jsonl'):\n",
+    "                jsonl_files.append(os.path.join(root, file))\n",
+    "    return jsonl_files\n",
+    "\n",
+    "# Function to merge multiple jsonl files into a single file \n",
+    "def merge_jsonl_files(directory, output_file):\n",
+    "    jsonl_files = list_jsonl_files(directory)\n",
+    "    \n",
+    "    with open(output_file, 'w') as outfile:\n",
+    "        for input_file in jsonl_files:\n",
+    "            with open(input_file, 'r') as infile:\n",
+    "                for line in infile:\n",
+    "                    try:\n",
+    "                        json_object = json.loads(line.strip())\n",
+    "                        json.dump(json_object, outfile)\n",
+    "                        outfile.write('\\n')\n",
+    "                    except json.JSONDecodeError:\n",
+    "                        print(f\"Skipping invalid JSON in {input_file}: {line.strip()}\")\n",
+    "\n",
+    "    print(f\"Merged {len(jsonl_files)} JSONL files into {output_file}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9bb0c80a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
```

Comment:
Also maybe worth mentioning that without explictly setting ``$NEMO_HOME``, it is set as `/root/.cache/nemo`